pandas/_typing.py

```diff
@@ -45,6 +45,7 @@
 # other
 
 Dtype = Union[str, np.dtype, "ExtensionDtype"]
+DtypeObj = Union[np.dtype, "ExtensionDtype"]
```

Comment:
shouldn't we call this DtypeType?

Reply:
DtypeType seems easy to confuse with dtype.type, like it would be the type for CategoricalDtypeType

Reply:
ahh i c, so this is not a type, just a non-string Dtype. ok then.